### PR TITLE
AGL GH Followers/ing

### LINKED ISSUE #29: User Dashboard: Github Following
👇 👇 👇 👇 
**REMINDER:** Don't start this story until you and your partner pair on and deploy the previous user story.
☝️ ☝️ ☝️ ☝️ 

**Details:** A user exists in the database. They have a Github token stored. Generate a token here for testing and development: https://github.com/settings/tokens

A future story will implement generating these tokens dynamically but that isn't necessary here.

```
As a logged in user
When I visit /dashboard
Then I should see a section for "Github"
And under that section I should see another section titled "Following"
And I should see list of users I follow with their handles linking to their Github profile
```

**API endpoint documentation**
https://developer.github.com/v3/users/followers/#list-users-followed-by-another-user

- [ ] API call uses authenticated call
- [ ] No hashes in the view.
- [ ] Uses objects in the view to represent the Github users.

### LINKED ISSUE #30: User Dashboard: Github Followers
🚔 👮‍♀️ 👮‍♀️ 👮‍♀️ 
**DO NOT START THIS STORY UNLESS THE STORY REFERENCED BELOW HAS BEEN COMPLETED BY BOTH IN YOUR PAIR.**

You should pair on story  "User Dashboard: Github Repositories" and deploy it prior to working on this one.

Continue this process for all stories labeled `spike first`. You will learn more this way and produce better code.

🚓 💨 👮 👮 👮 

**Details:** A user exists in the database. They have a Github token stored. Generate a token here for testing and development: https://github.com/settings/tokens

A future story will implement generating these tokens dynamically but that isn't necessary here.

No need to hit the endpoint for followings. A future story will address this.

```
As a logged in user
When I visit /dashboard
Then I should see a section for "Github"
And under that section I should see another section titled "Followers"
And I should see list of all followers with their handles linking to their Github profile
```

**API endpoint documentation**
https://developer.github.com/v3/users/followers/

- [ ] API call uses authenticated call: `/user/followers` instead of `/users/:username/followers`
- [ ] No hashes in the view.
- [ ] Uses objects in the view to represent the followers.

### PR DESCRIPTION
**Pull request by alex-latham**
Features:
* Adds Followers and Following sections to a user's dashboard if they have a `gh_token`

Controllers:
* Adds `delegate :gh_token` to ApplicationController to replace a `def gh_token` (Rubocop citation)
* Adds additional instance variables `@followers` and `@following` to Users#show
* Updates error message in sad path for creating a user

Models:
* New GitHub::User superclass (parent to GitHub::Follower and GitHub::Following) with readers `#login` and `#url`.
* GitHub::Follower and GitHub::Following each contain a single class method to list GitHub::Users

Services:
* GitHubServices now contains `#followers` and `#following` methods

Spec:
* Adds feature and model tests with edge case testing for GitHub::Repo, GitHub::User, GitHub::Follower, and GitHub::Following

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -3,13 +3,16 @@ class ApplicationController < ActionController::Base
...
   helper_method :tutorial_name
+  helper_method :gh_token
 
   add_flash_types :success
 
   def current_user
     @current_user ||= User.find(session[:user_id]) if session[:user_id]
   end
 
+  delegate :gh_token, to: :current_user
```
Comment (TravisBorgsmiller) [inline app/controllers/application_controller.rb:14]: rubocop for the win

File: app/models/git_hub/user.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+module GitHub
```
Comment (TravisBorgsmiller) [inline app/models/git_hub/user.rb:1]: I like this solution to dry up the follower and following code

**Review by TravisBorgsmiller — APPROVED** [review]
